htdocs/compta/paiement_charge.php

```diff
@@ -223,8 +223,8 @@
 	print '<tr><td class="tdtop">'.$langs->trans("RemainderToPay").'</td><td>'.price($total-$sumpaid,0,$outputlangs,1,-1,-1,$conf->currency).'</td></tr>';*/
 
 	print '<tr><td class="fieldrequired">'.$langs->trans("Date").'</td><td>';
-	$datepaye = dol_mktime(12, 0, 0, $_POST["remonth"], $_POST["reday"], $_POST["reyear"]);
-	$datepayment = empty($conf->global->MAIN_AUTOFILL_DATE) ? (empty($_POST["remonth"]) ?-1 : $datepaye) : 0;
+	$datepayment = dol_mktime(12, 0, 0, $_POST["remonth"], $_POST["reday"], $_POST["reyear"]);
+	$datepayment = ($datepayment == '' ? (empty($conf->global->MAIN_AUTOFILL_DATE) ?-1 : '') : $datepayment);
```

Comment:
If we submit the form and there is an error and MAIN_AUTOFILL_DATE is not set, the datepayment will always be set to -1 but it should remains in this cas the the value submitted so the form doesn't lose value entered. Right ?

Reply:
True, I've just copy paste what was done on other page with MAIN_AUTOFILL_DATE, and was thinking about rewrite all $_POST we GETPOST, but time is missing. For this one I've just do the correct way.